src/validation.cpp

```diff
@@ -967,7 +972,19 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
     CTxMemPool::setEntries collective_ancestors = *ancestors;
     collective_ancestors.merge(ws.m_ancestors_of_in_package_ancestors);
     if (const auto err{ApplyV3Rules(ws.m_ptx, collective_ancestors, ws.m_num_in_package_ancestors, ws.m_conflicts, ws.m_vsize)}) {
-        return state.Invalid(TxValidationResult::TX_MEMPOOL_POLICY, "v3-rule-violation", err->first);
+        // Disabled within package validation.
+        if (err->second != nullptr && !args.m_package_submission) {
```

Comment:
did you consider internalizing this logic inside `ApplyV3` directly to reduce conceptual sprawl

Reply:
Do you mean all of the sibling eviction logic? That doesn't work because we'd only enforce that the tx pays enough to evict siblings OR conflicting transactions; it needs to pay for both.

If you mean the logic adding stuff to `m_conflicts`, `m_iters_conflicting`, etc., it's because I don't like in-out parameters.

Reply:
the later, but at least you gave a reason why not

Reply:
Yeah I think passing those in would be a bit unhygienic. `ApplyV3Rules` would be using the contents beforehand to decide whether a child would be replaced, and then potentially modifying it in a way that looks like the child would replace it.